spec.html

```diff
@@ -39385,28 +39284,28 @@ <h1>GeneratorValidate ( _generator_ )</h1>
       </emu-clause>
 
       <emu-clause id="sec-generatorresume" aoid="GeneratorResume">
-        <h1>GeneratorResume ( _generator_, _value_ )</h1>
+        <h1>GeneratorResume ( _generator_, _value_ [ , _iterationKind_ ] )</h1>
         <p>The abstract operation GeneratorResume takes arguments _generator_ and _value_. It performs the following steps when called:</p>
         <emu-alg>
-          1. Let _state_ be ? GeneratorValidate(_generator_).
+          1. Let _state_ be ? GeneratorValidate(_generator_, _iterationKind_).
```

Comment:
If the operation has an optional argument, it can't just pass it on to another operation without first checking that it's "present" (and, if it isn't, setting it to some value).

However, there doesn't appear to be much reason for it to be optional. There's only one invocation of `GeneratorResume` where it isn't supplied, so just change that to supply some other value. (E.g., `~empty~`, or something more descriptive if that can be devised.) Then make `GeneratorValidate` switch on that rather that whether the arg is present.

(As for `GeneratorResumeAbrupt`, none of its invocations supply a 3rd arg, so it doesn't need its `_iteratorKind_` parameter at all.)